### PR TITLE
Utf8 & update source

### PR DESCRIPTION
**Pull request by RestingImmortal**
This PR Does as it says in the title, adds utf8 support as well as updates the source folder have the same features as base ES.

What does UTF8 do? It will allow the game to be translated into various different languages, or just simply display fancy characters like ĝ. 

As for the source updates, here is a full list:
Scram Drive is now displayed as a per second value, matching jump speed
Improved reparenting of carried ships, allowing them to be able to behave independently
The status overlay displays the threshold at which a ship will be disabled
Support for variable ammo quantity consumption
Scanning rings will again properly scale to the ship
Keyboard and Mouse reading is now handled in Engine rather than both AI and Engine
The fast forward key is now configurable
Creates conditions which are reserved and synced to the player's flagship's crew.
Limits pop-ups when multiple missions have on visit dialogues
Autopilot will now resume if it was interrupted by a movement key and the autopilot key is still held
Removes a possible divide-by-zero in the shipyard and outfitter
Map panels will correctly color active missions that cannot be completed
Allow customization of the government's displayed name
Require a new press to exit a conversation or select a choice
If not carried, fighters and drones are launched instead of sold
The display for a ship's total outfit space will now scale with installed outfits expansions
Print line numbers in DataNode traces
Call fclose on the error log
Escorts with more than one hyperdrive type can now determine if they have the necessary amount of fuel before their parent begins leaving the system
Injured, player-owned escorts regroup with the player's flagship
Only update AI with player commands when actively in-flight
Support nested phrase interpolation
More support for add/remove syntax
Window start and management is now a separate class
Added a command that will bring the flagship to a stop
Restore cursor-hiding in fullscreen
Check proper ship when using keyboard in Shipyard
Call CheckReferences even when no save exists
Only handle inputs given when the engine is full
Swap target parameter type of some AI methods from Ship to Body
Allow the UI to handle keys before checking Fast Forward, Fullscreen
Adds a solar heat mechanic
Update the automatic conditions daily
Add flagship System and Planet to player conditions
Account for the depreciation grace period in the max age of an outfit or ship
Negative repair costs no longer block repairing

This changes the required libraries, and I believe for windows it uses these: 
[win32-dev.zip](https://github.com/kikotheexile/Endless-Sky-Civil-War/files/4402040/win32-dev.zip)
[win64-dev.zip](https://github.com/kikotheexile/Endless-Sky-Civil-War/files/4402042/win64-dev.zip)

### CODE REVIEW

File: source/Cache.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/* Cache.cpp
+Copyright (c) 2018 by Michael Zahniser
```
Comment (Lorantine) [inline source/Cache.cpp:2]: This is a new file, correct? Update the copyright then. :)

File: source/Cache.h [hunk 2]
```diff
@@ -0,0 +1,330 @@
+/* Cache.h
+Copyright (c) 2018 by Michael Zahniser
```
Comment (Lorantine) [inline source/Cache.h:2]: This is a new file, correct? Update the copyright then. :)

File: source/GameWindow.cpp [hunk 3]
```diff
@@ -0,0 +1,372 @@
+/* GameWindow.cpp
+Copyright (c) 2014 by Michael Zahniser
```
Comment (Lorantine) [inline source/GameWindow.cpp:2]: This is a new file, correct? Update the copyright then. :)

File: source/GameWindow.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/* GameWindow.h
+Copyright (c) 2014 by Michael Zahniser
```
Comment (Lorantine) [inline source/GameWindow.h:2]: This is a new file, correct? Update the copyright then. :)

**Comment by Lorantine** [comment]
Unless browser Github was being... strange, or I'm mistaken, the 4 files I marked are new files, right?
If they already existed, just leave the copyright, else update the copyright (to either Kiko or just whoever wrote the file (you?)).

**Comment by kikotheexile** [comment]
we dont have to update the copyright file yet as we're literally just going to rip the one from ESC on the day of release then add our own additions to it. 

The source updates and additional support are much appreciated, and per your direction on github and my double checking right now, i see no conflicts with our own source updates unless im blind. Merging now.

File: source/Cache.cpp (hunk 1 above)
Comment (oo13) [reply source/Cache.cpp:2]: It's the first published year, and I would assign MZ the copyright of my part.

File: source/Cache.h (hunk 2 above)
Comment (oo13) [reply source/Cache.h:2]: It's the first published year, and I would assign MZ the copyright of my part.

**Pull request merged**